priv/tiff/pyramid.js

```diff
@@ -1,99 +1,70 @@
 const AWS = require("aws-sdk");
-const Sharp = require("sharp");
+const sharp = require("sharp");
 const URI = require("uri-js");
 const fs = require("fs");
-const stream = require("stream");
 const tempy = require("tempy");
 
-function createPyramidTiff(source, dest) {
-  return new Promise((resolve, reject) => {
-    try {
-      makeInputFile(source)
-        .catch(err => reject(err))
-        .then(inFile => {
-          console.error(`Creating pyramidal TIFF from ${inFile}`);
-          Sharp(inFile)
-            .limitInputPixels(false)
-            .resize({
-              width: 15000,
-              height: 15000,
-              fit: "inside",
-              withoutEnlargement: true
-            })
-            .tiff({
-              compression: "jpeg",
-              quality: 75,
-              tile: true,
-              tileHeight: 256,
-              tileWidth: 256,
-              pyramid: true
-            })
-            .toBuffer()
-            .then(data =>
-              sendToDestination(data, dest)
-                .then(resolve(dest))
-                .catch(err => reject(err))
-            )
-            .finally(() => {
-              console.error(`Deleting ${inFile}`);
-              fs.unlink(inFile, err => {
-                if (err) {
-                  console.error(err);
-                }
-              });
-            });
-        });
-    } catch (err) {
-      reject(err);
-    }
-  });
-}
+const createPyramidTiff = async (source, dest) => {
+  const inputFile = await makeInputFile(source);
+  const pyramidTiff = await sharp(inputFile)
+    .limitInputPixels(false)
+    .resize({
+      width: 15000,
+      height: 15000,
+      fit: "inside",
+      withoutEnlargement: true
+    })
+    .tiff({
+      compression: "jpeg",
+      quality: 75,
+      tile: true,
+      tileHeight: 256,
+      tileWidth: 256,
+      pyramid: true
+    })
+    .toBuffer();
+  await sendToDestination(pyramidTiff, dest);
+};
```

Comment:
Nice refactor.

Reply:
Thanks @adamjarling - I just noticed I missed the `finally` clause that unlinks the file over to the left on line 39. Looking for where to stick that in...

Reply:
Can you just wrap lines 9-26 in a `try { }` and then add a `finally { fs.unlink(inputFile) }` at the end?